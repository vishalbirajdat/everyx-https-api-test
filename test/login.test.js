const request = require("supertest");

const API_BASE_URL = process.env.API_BASE_URL || "http://localhost:8800";
const api = request(API_BASE_URL);

const ENDPOINT = "/auth/v2/login";
// Fixed testEmail declaration and assignment
const testEmail = "vishalbirajdar7030@gmail.com";

const normalizeResponse = (response) => {
  const stableBody = { ...response.body };
  if (stableBody.timestamp) stableBody.timestamp = "<DYNAMIC_TIMESTAMP>";
  if (stableBody.sessionId) stableBody.sessionId = "<DYNAMIC_SESSION_ID>";
  if (stableBody.message && typeof stableBody.message === "string") {
    // Optionally mask dynamic parts of the message
    stableBody.message = stableBody.message.replace(/to (.*)/, "to <DYNAMIC_EMAIL>");
  }
  return { status: response.status, body: stableBody };
};

describe("Auth V2 Login Email Link", () => {
  it("should send login link to valid email", async () => {
    const res = await api.get(`${ENDPOINT}/${encodeURIComponent(testEmail)}`);
    expect(normalizeResponse(res)).toMatchSnapshot("login-link-valid-email");
  });

  it("should fail for invalid email format", async () => {
    const invalidEmail = "notanemail";
    const res = await api.get(`${ENDPOINT}/${encodeURIComponent(invalidEmail)}`);
    expect(normalizeResponse(res)).toMatchSnapshot("login-link-invalid-email");
  });

  it("should fail for missing email param", async () => {
    const res = await api.get(`${ENDPOINT}/`);
    expect(normalizeResponse(res)).toMatchSnapshot("login-link-missing-email");
  });
});



// Taking time to send the testEmail
// linlk