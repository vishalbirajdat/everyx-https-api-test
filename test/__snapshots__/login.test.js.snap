// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Auth V2 Login Email Link should fail for invalid email format: login-link-invalid-email 1`] = `
{
  "body": {
    "error": "Bad Request",
    "message": ""email" must be a valid email",
    "statusCode": 400,
    "validation": {
      "keys": [
        "email",
      ],
      "source": "params",
    },
  },
  "status": 400,
}
`;

exports[`Auth V2 Login Email Link should fail for missing email param: login-link-missing-email 1`] = `
{
  "body": {},
  "status": 404,
}
`;

exports[`Auth V2 Login Email Link should send login link to valid email: login-link-valid-email 1`] = `
{
  "body": {
    "message": "Email sent successfully",
  },
  "status": 200,
}
`;
